pkg/api/steve/projectresources/handlers.go

```diff
@@ -0,0 +1,703 @@
+		for _, n := range nss {
+			_, ok := namespaces[n.Name]
+			if (op != notOp && ok) || (op == notOp && !ok) {
+				result[n.Name] = n
+			}
+		}
+		namespaces = result
+	} else {
+		// Either there was no selector so all namespaces in the project are valid, or
+		// the selector had the matching project so all namespaces in the project are
+		// included regardless of what namespaces were provided.
+		for _, n := range nss {
+			namespaces[n.Name] = n
+		}
+	}
+	resp, err := h.getResourcesForNamespaces(r, resource, resourceClient, namespaces)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	returnResp(w, resp)
+}
+
+// globalHandler handles admin requests for resources in any projects or any namespaces.
+func (h *handler) globalHandler(w http.ResponseWriter, r *http.Request) {
+	logrus.Tracef("[%s] handling request %s\n", apiServiceName, r.URL.Path)
+	vars := mux.Vars(r)
+	op := vars["op"]
+	var projectsOrNamespaces []string
+	if value, ok := vars[projectsOrNamespacesVar]; ok {
+		projectsOrNamespaces = strings.Split(value, ",")
+	}
+	resource, err := h.gvrFromVars(vars)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+	resourceClient := h.dynamicClient.Resource(resource)
+	projects, namespaces, err := h.projectsAndNamespaces(projectsOrNamespaces)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	// get all namespaces for projects
+	for _, p := range projects {
+		nss, err := h.namespaceCache.List(labels.Set{nslabels.ProjectIDFieldLabel: p}.AsSelector())
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		for _, n := range nss {
+			namespaces[n.Name] = n
+		}
+	}
+	// take the complement of all the namespaces provided in the query
+	if op == notOp {
+		allNamespaces, err := h.namespaceCache.List(labels.Everything())
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		complementNamespaces := make(map[string]*corev1.Namespace)
+		for _, n := range allNamespaces {
+			if _, ok := namespaces[n.Name]; !ok {
+				complementNamespaces[n.Name] = n
+			}
+		}
+		namespaces = complementNamespaces
+	}
+	resp, err := h.getResourcesForNamespaces(r, resource, resourceClient, namespaces)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	returnResp(w, resp)
+}
+
+// getResourcesForNamespaces concurrently retrieves resources of a given GVR for a slice of namespaces.
+func (h *handler) getResourcesForNamespaces(r *http.Request, resource schema.GroupVersionResource, resourceClient dynamic.NamespaceableResourceInterface, namespaces map[string]*corev1.Namespace) (*unstructured.UnstructuredList, error) {
+	results := make(chan unstructured.Unstructured)
+	eg, ctx := errgroup.WithContext(r.Context())
+	resourceCollection := make([]unstructured.Unstructured, 0)
+	latestResourceVersion := 0
```

Comment:
I'm not sure that I quite understand the purpose of this value. Near as I can tell, we are using this to return the last resource version from the various collections we retrieved. So if the list in ns1 gives 2 and the list in ns2 gives 4, we set this value to 2.

This value seems like it would not have much meaning to me. We are returning it to the user as though it is the resourceVersion of this collection, but in reality each list is independent of one another. In the above example, I could update the coollection of ns1 (producing a version of 3), but we would still return 4 since 4 is still higher than 3, making the user think the collection hasn't changed. If I'm understanding the use of this value correctly, we should probably use a sum here rather than a max - that way if any of the underlying values change than our return value changes too. Though that could get pretty large - let me know what thoughts you have.,